git.md

```diff
@@ -1,2 +1,125 @@
+
+## 1. Preserve shared history
+
+Shared commits should never change. Some Git commands preserve the history, but
+others don't. For example, `git commit --amend` and `git rebase --interactive`
+alter the git history; they help to overwrite  a messy commit history with
+a cleaner one; you should use them only before sharing your work. To cleanup
+shared history use instead `git revert`.  If you try to push altered history by
+accident, Git will stop you; to benefit from this protection  avoid `--force`
+(or `-f`) when pushing to shared repositories.
+
+## 2. Keep commits focused
+
+A commit should focus on related changes only. For example, fixing two
+different bugs should produce two separate commits. Small commits make it
+easier for other developers to understand the changes and roll them back if
+something went wrong. You can avoid sharing unwanted changes via .gitignore. If
+you can't succinctly describe what the commit is doing in a couple of words,
+then it is likely too big.
+
+See also:
+
+* [The right size of a PR](https://github.com/2DegreesInvesting/practices/discussions/3).
+* [.gitignore templates](https://github.com/github/gitignore).
+* Undoing things with Git ([video](https://youtu.be/dZOfEF19yDk),
+[lesson](https://coderefinery.github.io/git-intro/05-undoing)).
+
+## 3. Commit frequently
+
+Committing frequently keeps your commits small. Small commits are more likely
+to be focused; they are easier to review, and easier to merge because they
+offer fewer opportunities for conflicts.
+
+## 4. The work you share should be complete
+
+Each commit and feature should do what it promises, completely; and reverting
+it should undo what that commit or feature meant to do, also completely. If
+the goal of one commit requires changes in two files, that one commit should
+include changes to both files. Similarly, if the goal of a feature requires
+two kinds of changes, that feature should likely include two commits.
+
+## 5. Explain how to test the effect of the changes you made
+
+Testing your code is important, particularly when sharing it with
+others. Automated tests are the best but not the only way to test the effects
+of the changes you made. However you did it, explain how you ensured that the
+code behaves as you expect, and how someone else may reproduce your results.
+
+## 6. Structure each commit message like an email
+
+Like an email, commit messages have a subject and an optional body. The subject
+should start with a line summarizing your changes (aim for 50 characters or
+less). Use the imperative, present tense -- for example, "Fix typo",
+not "Fixed typo" or "Fixes typo". The message body, if present,
+should follow a blank line; wrap around 70 characters; and explain the
+motivation for the change ("why", not "what" or "how").
+
+See also:
+
+* [Example of a commit message](https://github.com/2DegreesInvesting/resources/issues/74).
+* <https://r-pkgs.org/git.html#commit-best-practices.>
+
+## 7. Agree on a workflow
```

Comment:
this should come with a caveat that we should actually agree on a workflow I think. As in - these are OUR best practices so shouldn't we just say: structure your work according to XYZ workflow?

Reply:
I'm a little conflicted. 

On the one hand I thing this doc should be opinionated. This supports @MonikaFu's idea of committing to one way of doing things, and suggests that @tposey28's "nice to have's may need to find another place. So it is now the time to take the discomfort of excluding some good ideas in the interest of keeping the minimum we can all live with.

And on the other hand, I know different teams and projects may have different needs, suggesting this document should be a flexible. I believe it's enough to say "agree on a workflow" as long as it's clearly documented. See for example <https://github.com/github/gitignore#contributing-workflow> -- it wouldn't be hard for a new comer to fit in.

Reply:
@maurolepore I see what you mean. Good point. At the same time I feel that for people who are less familiar with GitHub choosing a workflow themselves might be intimidating. My proposal taking into account your comment would be to agree on a 'default' workflow with an indication that teams are free to use a different workflow per project when they see a benefit. What do you think?

Reply:
Sounds like an awesome idea -- having a "default" best practice when the number of practices considered "best" is greater than one :)